TYPO3.Flow/Classes/TYPO3/Flow/Security/Context.php

```diff
@@ -837,7 +864,18 @@ public function getContextHash()
             $this->initialize();
         }
         if ($this->contextHash === null) {
-            $this->contextHash = md5(implode('|', array_keys($this->getRoles())));
+            $contextHashSoFar = implode('|', array_keys($this->getRoles()));
+
+            $this->withoutAuthorizationChecks(function () use (&$contextHashSoFar) {
+                foreach ($this->globalObjects as $globalObjectsRegisteredClassName) {
+                    $globalObject = $this->objectManager->get($globalObjectsRegisteredClassName);
```

Comment:
You mean check if it's a lazy loading proxy? AFAIS `objectManager->get()` always returns a real class instance, lazy loading only in effect for injected properties. Or am I wrong?